javascore/nativecoinIRC2/src/main/java/foundation/icon/btp/nativecoinIRC2/NativeCoinService.java

```diff
@@ -0,0 +1,583 @@
+    private final Address irc2;
+    private final String name;
+    private final String tokenName;
+    private final VarDB<NCSProperties> properties = Context.newVarDB("properties", NCSProperties.class);
+
+    //
+    private final OwnerManager ownerManager = new OwnerManagerImpl("owners");
+
+    //
+    private final ArrayDB<String> coinNames = Context.newArrayDB("coinNames", String.class);
+    private final BranchDB<String, DictDB<Address, Balance>> balances = Context.newBranchDB("balances", Balance.class);
+    private final DictDB<String, BigInteger> feeBalances = Context.newDictDB("feeBalances", BigInteger.class);
+    private final DictDB<BigInteger, TransferTransaction> transactions = Context.newDictDB("transactions", TransferTransaction.class);
+
+    public NativeCoinService(Address _bmc, Address _irc2, String _name, String _tokenName) {
+        bmc = _bmc;
+        BMCScoreInterface bmcInterface = new BMCScoreInterface(bmc);
+        //BTPAddress btpAddress = BTPAddress.valueOf(bmcInterface.getBtpAddress());
+        String btpAddr = (String) Context.call(bmcInterface._getAddress(), "getBtpAddress");
+        BTPAddress btpAddress = BTPAddress.valueOf(btpAddr);
+        net = btpAddress.net();
+        irc2 = _irc2;
+        name = _name;
+        tokenName = _tokenName;
+        coinNames.add(_tokenName);
+    }
+
+    public NCSProperties getProperties() {
+        return properties.getOrDefault(NCSProperties.DEFAULT);
+    }
+
+    public void setProperties(NCSProperties properties) {
+        this.properties.set(properties);
+    }
+
+    private boolean isRegistered(String name) {
+        if (tokenName.equals(name)) {
+            return true;
+        }
+        return false;
+    }
+
+    static void require(boolean condition, String message) {
+        if (!condition) {
+            throw NCSException.unknown(message);
+        }
+    }
+
+    @External(readonly = true)
+    public String[] coinNames() {
+        String[] names = new String[2];
+        names[0] = name;
+        names[1] = tokenName;
+        return names;
+    }
+
+    @External(readonly = true)
+    public Balance balanceOf(Address _owner, String _coinName) {
+        if (_owner.equals(Context.getAddress())) {
+            Balance balance = new Balance(BigInteger.ZERO, BigInteger.ZERO, BigInteger.ZERO);
+            balance.setRefundable(feeBalances.getOrDefault(_coinName, BigInteger.ZERO));
+            return balance;
+        } else {
+            return getBalance(_coinName, _owner);
+        }
+    }
+
+    @External
+    public void reclaim(String _coinName, BigInteger _value) {
+        require(_value.compareTo(BigInteger.ZERO) > 0, "_value must be positive");
+
+        Address owner = Context.getCaller();
+        Balance balance = getBalance(_coinName, owner);
+
+        if (balance.getUsable().compareTo(BigInteger.ZERO) > 0) {
+            balance.setRefundable(balance.getRefundable().add(balance.getUsable()));
```

Comment:
So after claim, all my usable balance added to refundable balance, and then if I want to transfer, I need to deposit again. Is it better to convert refundable balance to usable balance.

Reply:
so there will be a new method to claim usable balance?

Reply:
I mean add refundable balance to usable balance and then subtract usable balance by value user want to claim. I make  usable balance still remain, user don't need to deposit it again.

Reply:
i see, i think that's a better idea too. ill do it

Reply:
This is Done!